corehq/motech/generic_inbound/utils.py

```diff
@@ -141,3 +141,12 @@ def _revert_api_request_log(request_log):
     if request_log.status == RequestLog.Status.SUCCESS:
         request_log.status = RequestLog.Status.REVERTED
         request_log.save()
+
+
+def revert_api_request_from_form(form_id):
+    from corehq.motech.generic_inbound.models import ProcessingAttempt
+    try:
+        attempt = ProcessingAttempt.objects.get(xform_id=form_id)
+        _revert_api_request_log(attempt.log)
+    except ProcessingAttempt.DoesNotExist:
+        return
```

Comment:
I feel like there's probably a better way to handle this, but I'm not sure.

Reply:
Yep, that's right. I based it off just a couple comments under the ticket itself, notably:
> we could probably add a signal (or direct functional call) to the “archive” function that checks for a processing attempt associated with the form and respond accordingly if present

Signals seemed interesting but also like an easy way to accidentally create circular logic, and some research seemed to say that a direct function call is usually clearer.